src/containers/App/App.js

```diff
@@ -69,30 +69,36 @@ export /* istanbul ignore next */ class App extends Component {
             <Route path="*" component={Breadcrumbs} />
           </Header>
           <SideNav defaultExpanded expanded aria-label="Side navigation">
-            <SideNavHeader icon={<span />}>
-              <SideNavDetails title="Namespace">
-                <SideNavSwitcher
-                  labelText={namespace}
-                  onChange={event => {
-                    this.props.selectNamespace(event.target.value);
-                  }}
-                  options={namespaces}
-                />
-              </SideNavDetails>
-            </SideNavHeader>
             <SideNavItems>
-              <SideNavLink element={NavLink} icon={<span />} to="/pipelines">
-                Pipelines
-              </SideNavLink>
-              <SideNavLink element={NavLink} icon={<span />} to="/pipelineruns">
-                PipelineRuns
-              </SideNavLink>
-              <SideNavLink element={NavLink} icon={<span />} to="/tasks">
-                Tasks
-              </SideNavLink>
-              <SideNavLink element={NavLink} icon={<span />} to="/extensions">
-                Extensions
-              </SideNavLink>
+              <SideNavMenu title="Tekton">
+                <SideNavLink element={NavLink} icon={<span />} to="/pipelines">
+                  Pipelines
+                </SideNavLink>
+                <SideNavLink element={NavLink} icon={<span />} to="/tasks">
+                  Tasks
+                </SideNavLink>
+              </SideNavMenu>
+              <NamespacesDropdown
+                titleText="Namespace"
+                label="Namespace"
+                selectedItem={{ text: namespace }}
+                onChange={event => {
+                  this.props.selectNamespace(event.selectedItem.text);
+                }}
+              />
+              <SideNavMenu title="Extensions">
```

Comment:
We should either hide Extensions when there are none, or add an empty state. At the moment we end up with a non-interactive title on this section and no indication whether something broke, could be confusing to users.